## Assignment #1 DATA 730

This assignment tests that you have the course tech stack set up. If you do, you will be able to complete this notebook and turn-in your work by pushing to your assignment repository.

## Tech stack checklist

*Fill out the checklist below once you have completed each task:*

- [x] GitHub account

Change the cell below 👇 to a markdown cell and write in your GitHub account handle

jh82-mads


In [1]:
# uncomment the following line and execute this cell to demonstrate that you have installed the tidymodels package
library('tidymodels')

── Attaching packages ────────────────────────────────────── tidymodels 1.4.1 ──

✔ broom        1.0.13     ✔ recipes      1.3.3 
✔ dials        1.4.4      ✔ rsample      1.3.2 
✔ dplyr        1.2.1      ✔ tailor       0.1.0 
✔ ggplot2      4.0.3      ✔ tidyr        1.3.2 
✔ infer        1.1.0      ✔ tune         2.1.0 
✔ modeldata    1.5.1      ✔ workflows    1.3.0 
✔ parsnip      1.6.0      ✔ workflowsets 1.1.1 
✔ purrr        1.2.2      ✔ yardstick    1.4.0 

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ purrr::discard() masks scales::discard()
✖ dplyr::filter()  masks stats::filter()
✖ dplyr::lag()     masks stats::lag()
✖ recipes::step()  masks stats::step()



## Let's do some statistical inference!

Read [this article](https://www.theguardian.com/world/2002/jan/04/euro.eu2). 

Execute the code below to investigate whether the coin is fair...

In [2]:
# These functions simulate tossing a fair coin n_tosses number of times for n_iterations number of iterations

toss_fair_coin_n_times = function(n_tosses = 250) {
    result = ifelse(runif(n = n_tosses) < 0.5, 'heads', 'tails')
    result
}

repeat_the_experiment = function(n_iterations, n_tosses = 250) {
    n_heads_each_iteration = c()
    for (i in 1:n_iterations) {
        n_heads = sum(toss_fair_coin_n_times(n_tosses) == 'heads')
        n_heads_each_iteration = c(n_heads_each_iteration, n_heads)
    } 
    n_heads_each_iteration
}

In [3]:
# Let's do 10,000 experiments of 250 tosses and see how often our outcome deviates from what was actually observed (139 heads)

n_iterations = 10000
n_tosses = 250
observed_deviation = 139 - 125 # 139 heads observed, expected value for a fair coin of 125

n_heads_each_experiment = repeat_the_experiment(n_iterations, n_tosses)

x = mean(abs(n_heads_each_experiment - 125) >= observed_deviation)
paste0('With a fair coin we would expect a deviation as large or larger than what we observed ', x*100, '% of times')

[1] "With a fair coin we would expect a deviation as large or larger than what we observed 8.56% of times"

**Do you think the coin was fair?** (Answer below in a markdown cell.)

I believe that we dont have enough evidence to rule the coin as unfair.

Suppose we take the Null Hypothesis of these experiments to be:
>"There is no statistically significant difference between the outcomes of the belgian coin and a fair coin."

After simulating these experiments 10,000 times, we have found that the number of heads outcomes for a fair coin would be at least as extreme as the belgian coin 8.56% of times.
Using a conventional siginificance level of 0.05 (i.e. 5%), we observe that the threshold for statistical significance is lower than our observed p-value (8.56% > 5%).

Thus we should not reject the Null Hypothesis, and so we cannot declare with confidence that the coin is unfair.